database/import.go

```diff
@@ -240,118 +148,56 @@ func parseBigInt(s string) *big.Int {
 	return bi
 }
 
-func validateDbConsistency(importCount int, repo *repository.HeaderRepository) error {
-	if dbHeadersCount, _ := repo.GetHeadersCount(); dbHeadersCount != importCount {
-		return fmt.Errorf("database is not consistent with csv file")
-	}
-
-	return nil
-}
-
-// TODO hide SQLite-specific code behind some kind of abstraction.
-func getSQLitePragmaValues(db *sqlx.DB) (*SQLitePragmaValues, error) {
-	var pragmaValues SQLitePragmaValues
+func validateDbConsistency(importCount int, repo *sql.HeadersDb, db *sqlx.DB) error {
+	ctx := context.Background()
 
-	pragmaQueries := map[string]interface{}{
-		"synchronous":  &pragmaValues.Synchronous,
-		"journal_mode": &pragmaValues.JournalMode,
-		"cache_size":   &pragmaValues.CacheSize,
+	if dbHeadersCount, _ := repo.Count(ctx); dbHeadersCount != importCount {
+		return fmt.Errorf("database is not consistent with csv file, imported %d headers, number of headers in database %d", importCount, dbHeadersCount)
 	}
 
-	for pragmaName, target := range pragmaQueries {
-		query := fmt.Sprintf("PRAGMA %s", pragmaName)
-		err := db.QueryRow(query).Scan(target)
-		if err != nil {
-			return nil, err
-		}
+	if maxHeight, _ := repo.Height(ctx); maxHeight != importCount-1 {
+		return fmt.Errorf("database is not consistent with csv file, current maximum header height (%d) is different from imported headers number -1 (%d)", maxHeight, importCount)
 	}
 
-	return &pragmaValues, nil
-}
-
-// TODO hide SQLite-specific code behind some kind of abstraction.
-func modifySQLitePragmas(db *sqlx.DB) error {
-	pragmas := []string{
-		"PRAGMA synchronous = OFF;",
-		"PRAGMA journal_mode = MEMORY;",
-		"PRAGMA cache_size = 10000;",
+	if err := validateHeightUniqueness(db); err != nil {
+		return fmt.Errorf("database is not consistent with csv file, %w", err)
 	}
 
-	for _, pragma := range pragmas {
-		if _, err := db.Exec(pragma); err != nil {
-			return err
-		}
+	if err := validateRandomHeadersConsistency(repo, int32(importCount-1)); err != nil {
```

Comment:
I don't like this check. It want check the entire database so still don't give 100% confirmation that everything is correct, beside that I would rather check if we have [checkpoints](https://github.com/bitcoin-sv/pulse/blob/master/internal/chaincfg/params.go#L253) in database so at least we know if we have expected headers. But actually this is in scope of different task, so I would rather suggest to remove this check.

Reply:
Removed